utils/patientmatching/rule_based_patient_matcher.bal

```diff
@@ -0,0 +1,274 @@
+        MPIDbConfig|error dbConfig = self.getMPIConfigData(config);
+        if dbConfig is error {
+            return throwFHIRError("Error occurred while getting the configurations details for the database client from config.json file.",cause = dbConfig);
+        }
+        mysql:Client dbClient = check new (dbConfig.host, dbConfig.username, dbConfig.password, dbConfig.database, dbConfig.port);
+        return dbClient;
+    }
+
+    isolated function getMatchingPatients(r4:Patient patient, ConfigurationRecord config) returns error|r4:BundleEntry[] {
+        stream<record {}, sql:Error?>|error dbPatientStream = self.getMPIData(patient, check self.getPatientMatcherRuleData(config), config);
+        if dbPatientStream is error {
+            return throwFHIRError("Error occurred while getting the matching patients from MPI.",cause = dbPatientStream);
+        }
+        r4:BundleEntry[] patients = [];
+        map<anydata>[]|sql:Error patientArray = from record {} patientRecords in dbPatientStream
+            select patientRecords;
+        if patientArray is sql:Error {
+            return throwFHIRError("Error occurred while getting the matching patients from MPI.",cause = patientArray);
+        }
+        r4:BundleEntrySearch bundleEntrySearch = {
+            mode: "match",
+            score: 1.0
+        };
+        foreach int i in 0 ... patientArray.length()-1 {
+            r4:BundleEntry bundleEntry = {
+                'resource: patientArray[i],
+                search: bundleEntrySearch
+            };
+            patients.push(bundleEntry);
+        }
+        return patients;
+    }
+
+    isolated function getMPIData(r4:Patient patient, RulesRecord rulesTable, ConfigurationRecord config) returns stream<record {}, sql:Error?>|error {
+        sql:Client|error dbClient = self.getMpiDbClient(config);
+        if dbClient is error {
+            return throwFHIRError("Error occurred while getting the database client to access MPI.", cause = dbClient);
+        }
+        error|string qry = self.getSQLQuery(patient, rulesTable, config);
+        if qry is error {
+            return throwFHIRError("Error occurred while generating SQL query.", cause = qry);
+        }
+        sql:ParameterizedQuery queryString = ``;
+        queryString.strings = [qry];
+        stream<record {}, sql:Error?> resultStream = dbClient->query(queryString);
+        return resultStream;
+    }
+
+    isolated function getMPIConfigData(ConfigurationRecord config) returns MPIDbConfig|error {
+        json|error masterPatientIndexHost = config?.masterPatientIndexHost;
+        json|error masterPatientIndexPort = config?.masterPatientIndexPort;
+        json|error masterPatientIndexDb = config?.masterPatientIndexDb;
+        json|error masterPatientIndexDbUser = config?.masterPatientIndexDbUser;
+        json|error masterPatientIndexDbPassword = config?.masterPatientIndexDbPassword;
```

Comment:
Use the relevant type here instead of `json`

Reply:
Fixed in 6b572e1a82f29a631673acb2710d7f455ad43f82